buildconfig/CMake/SipQtTargetFunctions.cmake

```diff
@@ -0,0 +1,90 @@
+# Defines utility functions to help with generating
+# targets that produce a Python module from a set
+# of .sip definitions
+#
+
+#
+# brief: Add a module target to generate Python bindings
+# for a set of sip sources. The sources list should be a list of filenames
+# without a path. The .sip module is generated in the CMAKE_CURRENT_BINARY_DIR
+# keyword: MODULE_NAME The name of the final python module (without extension)
+# keyword: MODULE_OUTPUT_DIR The final destination of the built module optional
+# keyword: SIP_SRC_DIR The directory containing the given .sip files
+# keyword: SIP_SRCS A list of input .sip file paths
+# keyword: HEADER_DEPS A list of header files that are included
+#                      in the .sip files. These are set as dependencies on
+#                      the target.
+# keyword: INCLUDE_DIRS A list of additional target_include_directories
+# keyword: LINK_LIBS A list of additional target_link_libraries
+# keyword: PYQT_VERSION A single value indicating the version of PyQt
+#                       to compile against
+function ( mtd_add_sip_module )
+  find_file ( _sipmodule_template_path NAME sipqtmodule_template.sip.in
+    PATHS ${CMAKE_MODULE_PATH} )
+  if ( NOT _sipmodule_template_path )
+    message ( FATAL "Unable to find sipqtmodule_template.sip.in in cmake module path. Cannot continue." )
+  endif ()
+
+  set (options )
+  set (oneValueArgs MODULE_NAME TARGET_NAME MODULE_OUTPUT_DIR
+                    SIP_SRC_DIR PYQT_VERSION FOLDER )
+  set (multiValueArgs SIP_SRCS HEADER_DEPS INCLUDE_DIRS LINK_LIBS )
+  cmake_parse_arguments (PARSED "${options}" "${oneValueArgs}"
+                         "${multiValueArgs}" ${ARGN} )
```

Comment:
Missing space after `(`. I wonder if there is a clang-format for cmake files...

Reply:
Thanks! I wish there was!